crates/bin/pindexer/src/dex_ex/mod.rs

```diff
@@ -673,13 +674,13 @@ struct PairMetrics {
     liquidity_change: f64,
 }
 
-#[derive(Debug, Clone, Type)]
+#[derive(Debug, Clone, Type, serde::Serialize)]
 #[sqlx(type_name = "batch_swap_summary")]
 struct BatchSwapSummary {
-    asset_start: Vec<u8>,
-    asset_end: Vec<u8>,
-    input: BigDecimal,
-    output: BigDecimal,
+    asset_start: String,
+    asset_end: String,
+    input: String,
+    output: String,
```

Comment:
Similarly, casting to String for `input` and `output` discards meaningful typing. I think a better choice would be `Amount`, as done elsewhere in the code.